### PR TITLE
Fix/13952: Update gradient type when changing state government rules (BW->BY)

### PR DESCRIPTION
**Pull request by flxschmidt**
## Description
Fixes a bug, the the background color (gradient type) doesn't update, when changing state government rules (Bund, BW, BY)

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13952

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift [hunk 1]
```diff
@@ -156,6 +160,7 @@ class HealthCertifiedPersonCellModel {
 	let switchableHealthCertificates: OrderedDictionary<String, HealthCertificate>
 
 	let onTapToDelete: (() -> Void)?
+	var onUpdateGradientViewGradientType: ((GradientView.GradientType) -> Void)?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift:163]:
```suggestion
	var onUpdateGradientType: ((GradientView.GradientType) -> Void)?
```

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift [hunk 2]
```diff
@@ -829,6 +829,8 @@ class HealthCertificateService: HealthCertificateServiceServable {
 				person.mostRecentWalletInfoUpdateFailed = true
 				completion?()
 			}
+			
+			self.updateGradients()
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift:833]: Do we need this?

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift (hunk 1 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift]: I have adopted your suggestion. 👍

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift (hunk 2 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift:833]: Yes, after update the `dccWalletInfo`, wo have to trigger a recalculation of the `GradientType` in the `HealthCertificateService`.

**Review by naveeddotio — APPROVED** [review]

**Review by 30mar — APPROVED** [review]

**Pull request merged**